datafusion/sql/src/unparser/dialect.rs

```diff
@@ -95,4 +145,76 @@ impl Dialect for CustomDialect {
     fn identifier_quote_style(&self, _: &str) -> Option<char> {
         self.identifier_quote_style
     }
+
+    fn supports_nulls_first_in_sort(&self) -> bool {
+        self.supports_nulls_first_in_sort
+    }
+
+    fn use_timestamp_for_date64(&self) -> bool {
+        self.use_timestamp_for_date64
+    }
+
+    fn interval_style(&self) -> IntervalStyle {
+        self.interval_style
+    }
+}
+
+// create a CustomDialectBuilder
```

Comment:
I think this comment belongs on `CustomDialectBuilder::new()`

Also, can you please add documenation to this (nice) new builder? Specifically I think the struct itself should have an example of how to use it and each method has a docstring that explains what it does (perhaps with a link to the relevant method on `CustomDialect`